runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -1175,20 +1175,30 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          auto *method = std::get<0>(recv);
          int32_t cpIndex = std::get<1>(recv);
          bool unresolvedInCP;
-         J9Method *ramMethod = jitGetJ9MethodUsingIndex(fe->vmThread(), method->cp(), cpIndex);
-         unresolvedInCP = !ramMethod || !J9_BYTECODE_START_FROM_RAM_METHOD(ramMethod);
 
-            {
-            TR::VMAccessCriticalSection resolveStaticMethodRef(fe);
-            ramMethod = jitResolveStaticMethodRef(fe->vmThread(), method->cp(), cpIndex, J9_RESOLVE_FLAG_JIT_COMPILE_TIME);
-            }
+         // Call parent method
+         auto *resolvedMethod = static_cast<TR_ResolvedJ9Method *>(method->getResolvedStaticMethod(comp, cpIndex, &unresolvedInCP));
 
-            // Create a mirror right away
          TR_ResolvedJ9JITServerMethodInfo methodInfo;
-         if (ramMethod)
-            TR_ResolvedJ9JITServerMethod::createResolvedMethodFromJ9MethodMirror(methodInfo, (TR_OpaqueMethodBlock *) ramMethod, 0, method, fe, trMemory);
-
-         client->write(response, ramMethod, methodInfo, unresolvedInCP);
+         if (resolvedMethod) 
+            {
+            TR_ResolvedJ9JITServerMethod::packMethodInfo(methodInfo, resolvedMethod, fe);
+            client->write(
+               response,
+               resolvedMethod->getPersistentIdentifier(),
+               methodInfo,
+               unresolvedInCP
+               );
+            } 
+         else 
+            {
+            client->write(
+               response,
+               NULL,
+               methodInfo,
+               unresolvedInCP
+            );
+            }
```

Comment:
I just realized that there is an easy way to get rid of the if statement here and everywhere else where you do 2 client writes.
When you write `resolvedMethod->getPersistentIdentifier()` or `NULL`, just replace that with a tertiary
`resolvedMethod ? resolvedMethod->getPersistentIdentifier() : NULL`.
You don't even have to check if a resolved method exists before calling `packMethodInfo`, because it checks that internally.